src/code/cutscene_camera.c

```diff
@@ -353,28 +357,576 @@ void CutsceneCamera_Reset(void) {
+}
+
+s16 CutsceneCamera_Interp_Smooth(Vec3f* camPos, f32* camFov, s16* camRoll, CsCmdCamPoint* pointCmd,
+                                 CsCmdCamMisc* miscCmd, CutsceneCameraInterp* interpState) {
+    f32 lerp;
+    f32 tmp1;
+    f32 tmp2;
+
+    if (interpState->type != CS_CAM_INTERP_SMOOTH) {
+        // Initialize
+        interpState->type = CS_CAM_INTERP_SMOOTH;
+        interpState->waypoint = 0;
+        interpState->curFrame = 0;
+        interpState->duration = 1;
+        if (camPos != NULL) {
+            interpState->initPos.x = camPos->x;
+            interpState->initPos.y = camPos->y;
+            interpState->initPos.z = camPos->z;
+        }
+        if (camFov != NULL) {
+            interpState->initFov = *camFov;
+        }
+        if (camRoll != NULL) {
+            interpState->initRoll = *camRoll;
+        }
+    }
+
+    tmp2 = (((pointCmd->weight + 100) * (pointCmd->duration / 2)) +
+            (((pointCmd->weight + 100) / 2) * (pointCmd->duration & 1)));
+    if (pointCmd->duration < 2) {
+        lerp = 1.0f;
+    } else {
+        tmp1 = (f32)(pointCmd->weight - 100) / (pointCmd->duration - 1);
+        lerp = ((interpState->curFrame * tmp1) + 100.0f) / tmp2;
+    }
+
+    interpState->curFrame++;
+
+    if (camPos != NULL) {
+        camPos->x += (pointCmd->pos.x - interpState->initPos.x) * lerp;
+        camPos->y += (pointCmd->pos.y - interpState->initPos.y) * lerp;
+        camPos->z += (pointCmd->pos.z - interpState->initPos.z) * lerp;
+    }
+
+    if (camFov != NULL) {
+        *camFov += (miscCmd->fov - interpState->initFov) * lerp;
+    }
+
+    if (camRoll != NULL) {
+        s16 new_var;
+        s16 temp;
+
+        new_var = CAM_DEG_TO_BINANG(miscCmd->roll);
+
+        temp = (s16)(new_var - (s16)interpState->initRoll);
+
+        *camRoll += (s16)(temp * lerp);
+    }
+
+    if (interpState->curFrame >= pointCmd->duration) {
+        // Finished
+        interpState->type = CS_CAM_INTERP_OFF;
+        return 1;
+    }
+
+    return 0;
+}
+
+s16 CutsceneCamera_Interp_Geo(Vec3f* camPos, f32* camFov, s16* camRoll, CsCmdCamPoint* pointCmd, CsCmdCamMisc* miscCmd,
+                              CutsceneCameraInterp* interpState) {
+    VecGeo sp40;
+    f32 sp3C;
+    f32 tmp1;
+    f32 tmp2;
+
+    if (interpState->type != CS_CAM_INTERP_GEO) {
+        // Initialize
+        interpState->type = CS_CAM_INTERP_GEO;
+        interpState->waypoint = 0;
+        interpState->curFrame = 0;
+        interpState->duration = 1;
+        if (camPos != NULL) {
+            interpState->unk_20 = OLib_Vec3fDist(&interpState->curPos, camPos) * Math_FTanF(DEG_TO_RAD(*camFov));
+        }
+        if (camFov != NULL) {
+            interpState->initFov = *camFov;
+        }
+        if (camRoll != NULL) {
+            interpState->initRoll = *camRoll;
+        }
+    }
+
+    tmp2 = (((pointCmd->weight + 100) * (pointCmd->duration / 2)) +
+            (((pointCmd->weight + 100) / 2) * (pointCmd->duration & 1)));
```

Comment:
Matches?
```suggestion
    tmp2 = (((pointCmd->weight + 100) * (pointCmd->duration / 2)) +
            (((pointCmd->weight + 100) / 2) * (pointCmd->duration % 2)));
```

Reply:
This was actually brought up in a different comment:
https://github.com/zeldaret/mm/pull/1411#discussion_r1346787655

Basically, could rewrite `(pointCmd->duration & 1)` as `(s32)((u32)pointCmd->duration % 2)`

Reply:
Hmmm, that's ugly. What about `% 2U`? It can sometimes help. If it still require too many casts then just leave it as is

Reply:
That fixes the `(u32)` case, so `(s32)(pointCmd->duration % 2U)` would work. Is it worth it?

Reply:
I'm not sure if it is worth it